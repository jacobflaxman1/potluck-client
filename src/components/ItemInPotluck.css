.list-ul {
    display:flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: #05386B;
}

.taken-item {
    text-align: center;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
}