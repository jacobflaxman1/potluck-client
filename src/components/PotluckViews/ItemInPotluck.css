/* .list-ul {
    display:flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    color: #05386B;
} */

.taken-item {
    text-align: center;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    margin: 10px;
    padding: 3px;
}

.user-taking {
    font-weight: 400;
    font-size: large;
}

.not-taken {
    display: flex;
    flex-direction: row;
    align-content: center;
    justify-content: center;
    font-size: large;
}

.button-take {
    padding: 6px 20px;
    font-size: large;

}