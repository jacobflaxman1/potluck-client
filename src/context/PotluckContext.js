import React, { Component } from 'react'
import itemApiService from '../services/item-api-service'
import UserApiService from '../services/user-api-service'

const PotluckContext = React.createContext({
    potluck: [],
    items: [],
    error:null,
    authToken: null,
    users: [],
    adminUser: [],
    usersInPotluck: [],
    setEror: () => {},
    setPotluck: () => {},
    setItems: () => {},
    clearError: () => {},
    clearPotluck: () => {},
    saveAuthToken: () => {},
    clearAuthToken: () => {},
    addPostedPotluck: () => {},
    updateItem: () => {},
    getAllUserNames: () => {},
    getGuestUsersInPotluck: () => {},
    getAdminUserInPotluck: () => {},
    // LIST CONTEXT
    potluckList: [],
    user: '',
    setPotluckList: () => {},
    setUser: () => {},
    setShowForm: () => {},
    showForm: false,
    setExpanded: () => {},
    render: false,
    setRender: () => {},
    clearPotluckInList: () => {}
})

export default PotluckContext

export class PotluckProvider extends Component {
    state = {
        potluck: [],
        items: [],
        users: [],
        adminUser: [],
        usersInPotluck:[],
        error: null,
        authToken: null,
        showForm: false,
        //LIST STATE
        potluckList: [],
        user: '',
        render: false
    }

    setRender = () => {
        this.setState({ render: !this.state.render})
    }

    setShowForm = () => {
        this.setState({ showForm: !this.state.showForm })
    }


    setError = error => {
        console.error(error)
        this.setState({ error })
    }

    clearError = () => {
        this.setState({ error: null })
    }

    setPotluck = (potluck) => {
        this.setState({ potluck })
    }

    setItems = (items) => {
        this.setState({ items })
    }

    clearPotluck = () => {
        this.setPotluck([])
    }

    clearPotluckInList = (id) => {
        let newList = [...this.state.potluckList]
        let index = this.state.potluckList.findIndex(el => el.potluck_id === id)

        if(index !== -1) {
            newList.splice(index, 1)
            this.setState({potluckList: newList})
        }
        
    }

    saveAuthToken = (authToken) => {
        this.setState({ authToken })
    }

    clearAuthToken = () => {
        this.setState({ authToken: null })
    }

    addPostedPotluck = (newPotluck) => {
        this.setState({ potluckList: [...this.state.potluckList, {...newPotluck, expanded: true}] })
    }
    
    updateItem = (id) => {
        itemApiService.updateItem(id)
            .then((data) => {
                const newItemData = this.state.items.map((item) => {
                    if(id === item.item_id) {
                        return data
                    }
                    return item
                })
                this.setState({items: newItemData})
        })
    }

    getAllUserNames = () => {
        UserApiService.getUsers()
            .then((data) => {
                let userNames = data.rows.map(user => {
                    return user
                })
                this.setState({ users: userNames})
            })
    }
    getGuestUsersInPotluck = (id) => {
        console.log(id)
        UserApiService.getUsersByPotluck(id)
            .then((data) => {
                let userNames = data.guests.map(user => {
                    return user
                })
                this.setState({ usersInPotluck: userNames})
        })
    }
    getAdminUserInPotluck = (id) => {
        UserApiService.getUsersByPotluck(id) 
            .then((data) => {
                let adminUser = data.admin.map(ad => {
                    return ad
                })
                this.setState({ adminUser: adminUser})
            })
    }
    //LIST FUNCTIONS
    setPotluckList = potluckList => {
        const expandedPotluckList = potluckList.map(data => {
            return {...data, expanded: false}
        })

        this.setState({ potluckList: expandedPotluckList })
      }

    setUser = (user) => {
        this.setState({ user })
      }


    setExpanded = (id) => {
        const index = this.state.potluckList.findIndex(el => el.potluck_id === id)
        let newPotluckState = [...this.state.potluckList]

        newPotluckState[index] = {...newPotluckState[index], expanded: !newPotluckState[index].expanded}
        this.setState({ potluckList: newPotluckState })
    }

    render() {
        const value = {
            potluck: this.state.potluck,
            items: this.state.items,
            users: this.state.users,
            usersInPotluck: this.state.usersInPotluck,
            adminUser: this.state.adminUser,
            error: this.state.error,
            authToken: this.state.authToken,
            potluckList: this.state.potluckList,
            user: this.state.user,
            setItems: this.setItems,
            setError: this.setError,
            setPotluck: this.setPotluck,
            clearPotluck: this.clearPotluck,
            saveAuthToken: this.saveAuthToken,
            clearAuthToken: this.clearAuthToken,
            addPostedPotluck: this.addPostedPotluck,
            updateItem: this.updateItem,
            getAllUserNames: this.getAllUserNames,
            getGuestUsersInPotluck: this.getGuestUsersInPotluck,
            getAdminUserInPotluck: this.getAdminUserInPotluck,
            setPotluckList: this.setPotluckList,
            setUser: this.setUser,
            clearError: this.clearError,
            showForm: this.state.showForm,
            setShowForm: this.setShowForm,
            setExpanded: this.setExpanded,
            render: this.state.render,
            setRender: this.setRender,
            clearPotluckInList: this.clearPotluckInList

        }
        return(
            <PotluckContext.Provider value = {value}>
                {this.props.children}
            </PotluckContext.Provider>
        )
    }

}