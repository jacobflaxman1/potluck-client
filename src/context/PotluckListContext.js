import React, { Component } from 'react'

const PotluckListContext = React.createContext({
  potluckList: [],
  error: null,
  user: '',
  setError: () => {},
  clearError: () => {},
  setPotluckList: () => {},
  setUser: () => {}
})
export default PotluckListContext

export class PotluckListProvider extends Component {
  state = {
    potluckList: [],
    user: '',
    error: null,
  };

  setPotluckList = potluckList => {
    this.setState({ potluckList })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  setUser = (user) => {
    this.setState({ user })
  }

  render() {
    const value = {
      potluckList: this.state.potluckList,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setPotluckList: this.setPotluckList,
      setUser: this.setUser
    }
    return (
      <PotluckListContext.Provider value={value}>
        {this.props.children}
      </PotluckListContext.Provider>
    )
  }
}
