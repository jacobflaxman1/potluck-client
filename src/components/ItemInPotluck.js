import React, { useContext } from 'react'
import PotluckContext from '../context/PotluckContext'
import './ItemInPotluck.css'
export default function ItemInPotluck(props) {

    const context = useContext(PotluckContext)
    
    let eachItem = props.items.map((element, index) =>  {
        if(element.item_name === '') {
            return null
        }
        if(!element.taken) {
            return (
                <ul className = 'list-ul'>
                    <li key = {index}> 
                        <button onClick = {() => context.updateItem(element.item_id)}> Take Item </button> 
                        {element.item_name}
                    </li>
                </ul>
            ) 
        } 
        else {
            return(
                <ul className = 'list-ul'>
                    <li key = {index}>
                        <div className = 'taken-item'> {element.user_name} is taking </div>
                        {element.item_name}
                    </li>
                </ul>
            )
        }
    })
    return eachItem
}