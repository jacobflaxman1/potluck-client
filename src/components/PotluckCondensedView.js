import React, { Component } from 'react'
import PotluckContext from '../context/PotluckContext'
import './PotluckListItem.css'

export default class PotluckListItem extends Component {
    static contextType = PotluckContext

    render() {
        const { potluck } = this.props
        const { setExpanded } = this.context
        return (
                <div className = 'potluck-container' 
                    style={{ textDecoration: 'none', margin: '5px', color: 'black' }}>
                    <div className = 'expand' onClick = {() => setExpanded(potluck.potluck_id)}> {(potluck.expanded) ? <i className="arrow up"></i> : <i className="arrow down"></i>} </div>
                    <div className = 'PotluckTitle'> {potluck.potluck_name}</div>
                </div>
        )
    }
}
