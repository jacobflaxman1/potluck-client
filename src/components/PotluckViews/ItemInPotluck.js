import React, { useContext } from 'react'
import PotluckContext from '../../context/PotluckContext'
import './ItemInPotluck.css'
export default function ItemInPotluck(props) {

    const context = useContext(PotluckContext)
    
    let eachItem = props.items.map((element, index) =>  {
        if(element.item_name === '') {
            return null
        }
        if(!element.taken) {
            return (
                    <li key = {index} className = 'not-taken'> 
                        <button className = 'button-take' onClick = {() => context.updateItem(element.item_id)}> Bring {element.item_name}
                        </button> 
                    </li>
   
            ) 
        } 
        else {
            return(
                    <li key = {index}>
                        <div className = 'taken-item'> <span className = 'user-taking'> {element.user_name} </span> is bringing {element.item_name} </div>
                    </li>
            )
        }
    })
    return eachItem
}