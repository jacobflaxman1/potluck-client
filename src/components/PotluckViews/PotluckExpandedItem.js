import React, { Component } from 'react'
import PotluckContext from '../../context/PotluckContext'
import './PotluckExpanded.css'
import PotluckItemPage from '../../routes/PotluckItemPage'

export default class PotluckListItem extends Component {
    static contextType = PotluckContext

    render() {
        const { potluck } = this.props
        const { setExpanded } = this.context
        return (
                <div className = 'potluck-container' 
                    style={{ textDecoration: 'none', margin: '5px', color: 'black' }}>
                    <div className = 'expand' onClick = {() => setExpanded(potluck.potluck_id)} > {(potluck.expanded) ? <i className="arrow up"></i> : <i className="arrow down"></i>} </div>
                         <PotluckItemPage className = 'expanded-content' potluck_id ={potluck.potluck_id} expandContent = {this.expandContent}/>
                </div>
        )
    }
}

