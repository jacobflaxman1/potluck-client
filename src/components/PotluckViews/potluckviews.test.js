import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { PotluckProvider } from '../../context/PotluckContext'
import PotluckExpandedItem from './PotluckExpandedItem';
import PotluckCondensedView from './PotluckCondensedView'
import ItemInPotluck from './ItemInPotluck'

describe(`Adds all of the potluck views to dom without crashing`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(
        <PotluckProvider>
            <PotluckExpandedItem />
        </PotluckProvider>
    )
    .find('.potluck-container')
    expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders without crashing', () => {
        const wrapper = shallow(
        <PotluckProvider>
            <PotluckCondensedView />
        </PotluckProvider>
    )
    .find('.potluck-container')
    expect(toJson(wrapper)).toMatchSnapshot()
    })
    
    it('renders without crashing', () => {
        const wrapper = shallow(
        <PotluckProvider>
            <ItemInPotluck />
        </PotluckProvider>
    )
    .find('.button-take')
    expect(toJson(wrapper)).toMatchSnapshot()
    })


})