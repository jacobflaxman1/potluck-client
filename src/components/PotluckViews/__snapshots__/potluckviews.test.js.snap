// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds Potluck Form to dom without crashing renders without crashing 1`] = `null`;

exports[`Adds Potluck Form to dom without crashing renders without crashing 2`] = `null`;

exports[`Adds Potluck Form to dom without crashing renders without crashing 3`] = `null`;

exports[`Adds all of the potluck views to dom without crashing renders without crashing 1`] = `null`;

exports[`Adds all of the potluck views to dom without crashing renders without crashing 2`] = `null`;

exports[`Adds all of the potluck views to dom without crashing renders without crashing 3`] = `null`;
